### PR TITLE
fix: Sales funnel data is inconsistent

### PR DESCRIPTION
**Pull request by anupamvs**
Issue -  Funnel showing inconsistent data.
image

Fix - 
| Sales Stage  | Count Based On |
| ------------- | ------------- |
| Active Lead  | Lead status in (**Open**, **Lead**,  **Replied**) |
| Opportunities  | Opportunity status in (**Open**,  **Replied**) , only from Lead|
| Quotations  | Quotation status in (**Open**,  **Replied**) and Doc is **Submitted** and Only from Lead/Opportunity |
| Converted  | Quotation status is **Ordered** and Doc is **Submitted**  |

image

Also removed Sales Orders as the duplicacy gave us no proper count

### CODE REVIEW

**Comment by anupamvs** [comment]
@mergifyio backport version-12-hotfix

File: erpnext/selling/page/sales_funnel/sales_funnel.py [hunk 1]
```diff
@@ -38,11 +34,16 @@ def get_funnel_data(from_date, to_date, company):
...
+		and (date(contact.creation) between %s and %s) and status != "Passive" """, (from_date, to_date))[0][0]
+
 	return [
-		{ "title": _("Active Leads / Customers"), "value": active_leads, "color": "#B03B46" },
+		{ "title": _("Active Leads"), "value": active_leads, "color": "#B03B46" },
 		{ "title": _("Opportunities"), "value": opportunities, "color": "#F09C00" },
 		{ "title": _("Quotations"), "value": quotations, "color": "#006685" },
-		{ "title": _("Sales Orders"), "value": sales_orders, "color": "#00AD65" }
+		{ "title": _("Sales Orders"), "value": sales_orders, "color": "#00AD65" },
+		{ "title": _("Converted"), "value": converted, "color": "#228B22" },
```
Comment (marination) [inline erpnext/selling/page/sales_funnel/sales_funnel.py]:
```suggestion
		{ "title": _("Converted"), "value": converted, "color": "#228B22" }
```

**Review by marination — DISMISSED** [review]

File: erpnext/selling/page/sales_funnel/sales_funnel.py [hunk 2]
```diff
@@ -20,29 +20,27 @@ def get_funnel_data(from_date, to_date, company):
...
+		and opportunity_from='Lead' and company=%s""", (from_date, to_date, company))[0][0]
 
 	quotations = frappe.db.sql("""select count(*) from `tabQuotation`
 		where docstatus = 1 and (date(`creation`) between %s and %s)
-		and status != "Lost" and company=%s""", (from_date, to_date, company))[0][0]
+		and (opportunity!="" or quotation_to="Lead") and company=%s""", (from_date, to_date, company))[0][0]
+
+	converted = frappe.db.sql("""select count(*) from `tabQuotation`
+		where docstatus = 1 and (date(`modified`) between %s and %s)
+		and status="Ordered" and company=%s""", (from_date, to_date, company))[0][0]
```
Comment (marination) [inline erpnext/selling/page/sales_funnel/sales_funnel.py:36]: struck me later, get count of customers where `lead_name` (From Lead) is not empty.

Comment (marination) [inline erpnext/selling/page/sales_funnel/sales_funnel.py]: struck me later, get count of customers where `lead_name` (From Lead) is not empty, also creation date should lie between the from and to date. So Customers converted from Leads between the concerned dates

**Review by marination — APPROVED** [review]

**Pull request merged**